### ISSUE TITLE
couldn't read file "/mingw64/bin/gitk"

### ISSUE THREAD

**Issue by lolive** [git-for-windows/git#533]
Hello.

I run Git-Bash version 2.6.2 on Windows 7.
When I try to run gitk, i get this error message: 
couldn't read file "/mingw64/bin/gitk": no such file or directory.

When I try to circunvent this problem with this command line:
cat /mingw64/bin/gitk |/usr/bin/bash

I get this error:
couldn't read file "/usr/bin/bash": no such file or directory.

This bug was reproduced on my colleague's computer (same Windows 7, same git bash).

Is it a known issue?

**Comment by dscho** [comment]
Whoa. I have *no* clue how you tried to run gitk. All I know is that it works *perfectly* here, when I call it

* from the start menu,
* from Git Bash, and
* from Git CMD

Maybe you want to follow http://whathaveyoutried.com to give me a prayer of reproducing your issue here?

**Comment by lolive** [comment]
I have tried to run the various commands I mentionned from the Git Bash command line. 
I agree that this is a very weird issue.

**Comment by dscho** [comment]
@lolive well, repeating what you already said will not provide the necessary information about to help you.

In particular, I challenge you to really *read* the page I linked to, and follow the advice detailed there. This should result in a much, much more detailed report.

FWIW I tried a couple of things *here* (and yeah, as you probably guessed by now, this problem seems to be limited very much to your setup, it works perfectly here):

```sh
me@work MINGW64 /c/windows/system32
$ gitk

me@work MINGW64 /c/windows/system32
$ type gitk
gitk is hashed (/mingw64/bin/gitk)

me@work MINGW64 /c/windows/system32
$ ls -l $(which gitk)
-rwxr-xr-x 1 johasc 1049089 349016 Oct 19 22:05 /mingw64/bin/gitk*
```

Maybe you fiddle with `PATH` (and failed to mention that) or maybe you are running mixed architectures, or somehow you are running this in Git for Windows 1.x' Bash or some such.

**Comment by thynameisblair** [comment]
Hello!

I stumbled across this issue myself, and I'm not too sure how to solve it. Let me tell you what I know:

- Running a freshly installed 64-bit version (2.14.1)
- Running `gitk` from Git Bash fails with a dialog box containing the message `couldn't read file "/mingw64/bin/gitk": no such file or directory`
- Running `git gui`, then selecting `Repository -> Vizualize master's history` _**does**_ correctly launch `gitk`

The difference appears to be due to path conversion: when examining with Process Explorer, a failed attempt to run `gitk` results in this process tree:
```
"C:\Program Files\Git\usr\bin\sh.exe"
    "C:\Program Files\Git\mingw64\bin\wish.exe" /mingw64/bin/gitk --
```
Whereas the successful run looks like:
```
"C:\Program Files\Git\mingw64\bin\wish.exe" "C:\Program Files\Git\mingw64/libexec/git-core\git-gui" --
    "C:\Program Files\Git\mingw64\bin\wish.exe" "C:/Program Files/Git/mingw64/bin/gitk" master -- --
```
Note that the path passed to `wish` is a proper Windows path in the working example.

I'm running Windows 10 build 1703, and I have Developer mode enabled; I checked the "Use symbolic links" option during installation of Git.

Let me know if this helps, or if I can provide any further information!
